hal/src/rtl872x/sleep_hal.cpp

```diff
@@ -88,6 +88,55 @@ class SleepClass {
         return SYSTEM_ERROR_NONE;
     }
 
+#if HAL_PLATFORM_IO_EXTENSION && MODULE_FUNCTION != MOD_FUNC_BOOTLOADER
+#if HAL_PLATFORM_MCP23S17
+    void configGpioWakeupSourceExt(const hal_wakeup_source_base_t* wakeupSources, hal_sleep_mode_t sleepMode) {
+        hal_pin_info_t* halPinMap = hal_pin_map();
+        auto source = wakeupSources;
+        bool config = false;
+        uint8_t gpIntEn[2] = {0, 0};
+        uint8_t intCon[2] = {0, 0};
+        uint8_t defVal[2] = {0, 0};
+
+        while (source) {
+            if (source->type == HAL_WAKEUP_SOURCE_TYPE_GPIO) {
+                auto gpioWakeup = reinterpret_cast<const hal_wakeup_source_gpio_t*>(source);
+                if (halPinMap[gpioWakeup->pin].type == HAL_PIN_TYPE_IO_EXPANDER) {
+                    config = true;
+                    switch(gpioWakeup->mode) {
+                        case RISING: {
+                            intCon[halPinMap[gpioWakeup->pin].gpio_port] |= (0x01 << halPinMap[gpioWakeup->pin].gpio_pin);
+                            break;
+                        }
+                        case FALLING: {
+                            intCon[halPinMap[gpioWakeup->pin].gpio_port] |= (0x01 << halPinMap[gpioWakeup->pin].gpio_pin);
+                            defVal[halPinMap[gpioWakeup->pin].gpio_port] |= (0x01 << halPinMap[gpioWakeup->pin].gpio_pin);
+                            break;
+                        }
+                        case CHANGE:
+                        default: {
+                            break;
+                        }
+                    }
+                    gpIntEn[halPinMap[gpioWakeup->pin].gpio_port] |= (0x01 << halPinMap[gpioWakeup->pin].gpio_pin);
+                }
+            }
+            source = source->next;
+        }
+
+        if (config) {
+            // TODO: check the returned result
```

Comment:
There's no return value from `Mcp23s17::writeRegister` other than `SYSTEM_ERROR_NONE` so this can be removed. There doesn't seem to be any meaningful error recovery anyway since this function is `void` return.

Same thing in the nRF copy of this code

Reply:
If this function can't be shared/de-deuped between the nRF version of it, add a comment above the function that these two versions are coupled